Lib/test/test_threading.py

```diff
@@ -2103,6 +2103,63 @@ def test__all__(self):
         support.check__all__(self, threading, ('threading', '_thread'),
                              extra=extra, not_exported=not_exported)
 
+    def test_set_name(self):
+        try:
+            get_name = _thread._get_name
+            _thread.set_name
+        except AttributeError:
+            self.skipTest("need thread._get_name() and thread.set_name()")
+
+        def work():
+            nonlocal work_name
+            work_name = get_name()
+
+        # set_name() limit in bytes
+        truncate = getattr(_thread, "_NAME_MAXLEN", None)
+        limit = truncate or 100
+
+        def create_test(name):
+            if sys.platform.startswith("solaris"):
+                encoding = "utf-8"
+            else:
+                encoding = sys.getfilesystemencoding()
+            encoded = name.encode(encoding, "replace")
+            if truncate is not None:
+                expected = os.fsdecode(encoded[:truncate])
+            else:
+                expected = os.fsdecode(encoded)
+            return (name, expected)
+
+        tests = [
+            # test short ASCII name
+            create_test("CustomName"),
```

Comment:
This is just a nitpick, and you can ignore it, but I would simply iterate a list of names and calculated the `expected` value in the body of the loop.
```
for name in tests:
    expected = ...
```